### ISSUE TITLE
Avoid system GDAL dependency

### ISSUE THREAD

**Issue by cmcaine** [IntegratedTransportPlanning/gcvt#143]
At the moment we depend on `ogr2ogr` (part of GDAL) to convert shp to geojson in the data processing pipeline. We can use a Julia GDAL wrapper to have Pkg manage that dependency for us, but we'll have to implement the bits of ogr2ogr we need ourselves.

This issue has some of the bits we'd need to do that: https://github.com/yeesian/ArchGDAL.jl/issues/147

**Comment by bovine3dom** [comment]
Why do we need to reimplement bits of `ogr2ogr`?

It looks like `GDAL_jll` comes with it:

https://github.com/JuliaPackaging/Yggdrasil/blob/27836590f0c0d28146c0010de9a485531a2e9327/G/GDAL/build_tarballs.jl#L92

**Comment by cmcaine** [comment]
Oh, neat. I don't know how we get at those binaries. Could you look into it
sometime?

**Comment by bovine3dom** [comment]
Yeah, no problem.

https://docs.binarybuilder.org/dev/jll/#ExecutableProduct looks pretty straightforward.

**Comment by bovine3dom** [comment]
This gets us 90% of the way there

```julia
]add GDAL_jll
using GDAL_jll
GDAL_jll.ogr2ogr_path() do exe
    run(`$exe --long-usage`)
end
```

I'll look at bunging it in `od-vis`.

**Comment by cmcaine** [comment]
I moved the geom processing into a function in pct.jl in this repo. Probably a better place to modify

The docs seem to suggest that this would work, which looks nicer to me:

```jl
]add GDAL_jll
using GDAL_jll
GDAL_jll.ogr2ogr() do exe
    run(`$exe --long-usage`)
end
```

**Comment by bovine3dom** [comment]
```
julia> using GDAL_jll
julia> GDAL_jll.ogr2ogr
ERROR: UndefVarError: ogr2ogr not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at ./Base.jl:26
 [2] top-level scope at REPL[2]:1
```

I think you're misreading the source.

```
ExecutableProduct(binname, varname::Symbol, dir_path="bin")
```

`binname` is the filename of the executable. `varname` is the ~~variable~~ function it gets stored in.

**Comment by cmcaine** [comment]
I was just reading the ExecutableProduct docs, which say that a _path will be made for you, so I think ogr2ogr_path_path probably exists, which seems redundant :)

**Comment by bovine3dom** [comment]
```
julia> GDAL_jll.ogr2ogr_path_path
"/home/olie/.julia/artifacts/213206980a92c153d0629256d187a69b6cce6efa/bin/ogr2ogr"
```

It _does_ exist and has brightened my day.

It's unexported, though.

**Comment by bovine3dom** [comment]
Closed in 0cfd3a7
